stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java

```diff
@@ -482,17 +482,26 @@ default int totalPackets() {
 
     /**
      * Send the packet but check to see if an error in the HTTP 4XX range was encountered so that we know if we
-     * should stop retrying
+     * should stop retrying. We should stop retrying if the reject packet had an F family error, and that error is not
+     * and F08. When we receive an F08 reject packet, we try to send smaller packets, so this should not short circuit.
      * @param preparePacket
+     * @param streamPacket
      * @return the returned response packet
      */
     @VisibleForTesting
-    protected InterledgerResponsePacket sendPacketAndCheckForFailure(InterledgerPreparePacket preparePacket) {
+    protected InterledgerResponsePacket sendPacketAndCheckForFailure(InterledgerPreparePacket preparePacket,
+                                                                     StreamPacket streamPacket) {
       InterledgerResponsePacket response = link.sendPacket(preparePacket);
-      response.handle((fulfill) -> {}, (reject) -> {
-        if (reject.getCode().equals(F00_BAD_REQUEST)) {
-          unrecoverableErrorEncountered.set(true);
-        }
+      response.handle(
+        (fulfill) -> {},
+        (reject) -> {
+          if (!reject.getCode().equals(F08_AMOUNT_TOO_LARGE) &&
```

Comment:
Shouldn't this also keep retrying on F99 errors?

Reply:
@sappenin would know better...  

The Java receiver only returns an F99 if the prepare amount < stream amount, but the Java sender doesn't have any retry behavior to deal with F99s.  Maybe it should?

Reply:
There are other cases where the JavaScript STREAM receiver returns F99 errors, too, which is something to consider.

> The Java receiver only returns an F99 if the prepare amount < stream amount

But should the payment fail if only a single packet doesn't meet the minimum exchange rate? Or should there be multiple attempts?

Rust and JS retry like normal on F99s, but I think we probably need more sophisticated behavior for the exchange rate issues.

Reply:
Agreed. For now, our main motivation for short circuiting is to reduce our log output/error alerting, which is mainly happening on F02s. I will narrow this condition to only short circuit on F02s, and we can discuss what we want to do on F99s.